Docs: Update alert data model of webhook-notifier documentation

hoptical opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**
Updates [alert data model of webook-notifer](https://grafana.com/docs/grafana/latest/alerting/manage-notifications/webhook-notifier/#alert) documentation. There are some fields in the [alert data model](https://github.com/grafana/alerting/blob/793c67215ba14d6515b93707caeb7e6d9bf82189/templates/template_data.go#L29) which are not listed in the documentation:
- values
- imageURL
- embeddedImage

Additionally, the `valueString` field has been marked as being deprecated in Grafana 10.  

**Why do we need this feature?**

Webhook developers need to have the full JSON documentation of the Alerting data model.

**Who is this feature for?**

Webhook developers


<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

**Special notes for your reviewer:**

I have considered `values` as an object. Please check if it's correct and if you're Ok with the deprecating note on the `valueString` field.

Please check that:
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

hoptical committed:
Add missed fields to webhook-notifier.md

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=67956) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=67956) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=67956) it.</sub>

yuri-tceretian commented on docs/sources/alerting/manage-notifications/webhook-notifier.md:
```diff
@@ -110,12 +110,15 @@ Example JSON body:
 | dashboardURL | string | **Will be deprecated soon**                                                        |
 | panelURL     | string | **Will be deprecated soon**                                                        |
+| imageURL     | string | URL of the screenshots of the panel alert rule assigned to notifications           |
+| embeddedImage| string | embedded screenshots of the panel alert rule assigned to notifications             |
```
embeddedImage are not populated in webhook. it is used only by email integration

hoptical replied:
Ok, I will remove it.

hoptical replied:
Done.

yuri-tceretian commented on docs/sources/alerting/manage-notifications/webhook-notifier.md:
```diff
@@ -110,12 +110,15 @@ Example JSON body:
 | startsAt     | string | Start time of the alert                                                            |
 | endsAt       | string | End time of the alert, default value when not resolved is `0001-01-01T00:00:00Z`   |
-| valueString  | string | Values that triggered the current status                                           |
+| valueString  | string | **Will be deprecated in Grafana 10**                                               |  
```
Oh, I found TODO that we missed https://github.com/grafana/alerting/blob/e067f119be06c7ae1e225d8b4bfc2c832a9b23e3/templates/template_data.go#L129 @grobinson-grafana should we delete it now?

hoptical replied:
Waiting for the response.

yuri-tceretian replied:
Let's delete `valueString` from the doc at all. We will delete it in code after that.

yuri-tceretian commented on docs/sources/alerting/manage-notifications/webhook-notifier.md:
```diff
@@ -110,12 +110,15 @@ Example JSON body:
 | silenceURL   | string | URL to silence the alert rule in the Grafana UI                                    |
 | dashboardURL | string | **Will be deprecated soon**                                                        |
 | panelURL     | string | **Will be deprecated soon**                                                        |
+| imageURL     | string | URL of the screenshots of the panel alert rule assigned to notifications           |
```
I would rephrase it. It is `URL of a screenshot of a panel assigned to the rule that created this notification`

hoptical replied:
I will rephrase it.

hoptical replied:
Done.

hoptical committed:
Merge branch 'grafana:main' into docs_alerting_webhook

hoptical committed:
Update webhook-notifier.md

Remove `imageEmbedded` field
Improve description of the `imageURL field
Add `values` to the example JSON body

hoptical commented on docs/sources/alerting/manage-notifications/webhook-notifier.md:
```diff
@@ -41,7 +41,10 @@ Example JSON body:
       "dashboardURL": "",
       "panelURL": "",
-      "valueString": "[ metric='' labels={} value=14151.331895396988 ]"
+      "values": {
```
@yuri-tceretian 
I have also added `values` field to the example json body.

hoptical committed:
Remove valuestring from webhook-notifier.md

yuri-tceretian commented on docs/sources/alerting/manage-notifications/webhook-notifier.md:
```diff
@@ -41,7 +41,9 @@ Example JSON body:
       "dashboardURL": "",
       "panelURL": "",
-      "valueString": "[ metric='' labels={} value=14151.331895396988 ]"
+      "values": {
+        "data-avg": 14151.331895396988
+      }
```
```suggestion
        "values": {
        "B": 44.23943737541908,
        "C": 1
      }
```
I think this example will be more intuitive. Because not many users rename their queries\expressions

yuri-tceretian commented on docs/sources/alerting/manage-notifications/webhook-notifier.md:
```diff
@@ -62,7 +64,9 @@ Example JSON body:
       "dashboardURL": "",
       "panelURL": "",
-      "valueString": "[ metric='' labels={} value=47043.702386305304 ]"
+      "values": {
+        "data-avg": 47043.702386305304
+      }
```
```suggestion
      "values": {
        "B": 44.23943737541908,
        "C": 1
      }
```

yuri-tceretian commented:
Please ` yarn run prettier:write` to make linter happy.

yuri-tceretian reviewed (approved):
LGTM

hoptical committed:
Update docs/sources/alerting/manage-notifications/webhook-notifier.md

Co-authored-by: Yuri Tseretyan <tceretian@gmail.com>

hoptical committed:
Update docs/sources/alerting/manage-notifications/webhook-notifier.md

Co-authored-by: Yuri Tseretyan <tceretian@gmail.com>

hoptical committed:
lint the webhook-notifier.md doc

hoptical commented:
All suggestions and linting applied.
